sunkit_image/coalignment/interface.py

```diff
@@ -0,0 +1,181 @@
+import warnings
+from typing import NamedTuple
+
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+from sunpy.util.exceptions import SunpyUserWarning
+
+from sunkit_image.coalignment.util.decorators import registered_methods
+
+__all__ = ["coalignment", "affine_params"]
+
+
+class affine_params(NamedTuple):
+    """
+    A named tuple to store the affine transformation parameters used for
+    updating the metadata.
+
+    Attributes
+    ----------
+    scale : tuple[tuple[float, float], tuple[float, float]]
+        The scale matrix representing the scaling transformation.
+        This 2x2 matrix defines how the image is scaled along the x and y axes.
+
+    rotation_matrix : tuple[tuple[float, float], tuple[float, float]]
+        The rotation matrix representing the rotation transformation.
+        This 2x2 matrix defines the rotation of the image in the plane.
+
+    translation : tuple[float, float]
+        The translation vector representing the translation transformation.
+        This 2-element tuple defines the shift of the image along the x and y axes in pixels.
```

Comment:
This documents the high level but I think Cadair suggested adding one docstring per attribute instead.